C@E Prefetch/prefetch_hls_fastly/src/main.rs

```diff
@@ -27,57 +24,60 @@ fn main(mut req: Request) -> Result<Response, Error> {
     let req_clone = req.clone_without_body();
     let req_clone_for_match = req.clone_without_body();
 
-    match (req_clone_for_match.get_method(), req_clone_for_match.get_path()) {
+    match (
+        req_clone_for_match.get_method(),
+        req_clone_for_match.get_path(),
+    ) {
         //When we get the media playlist we will also prefetch a few segments but also modify the playlist
         // to include the media playlist file name as a query param (instead of using cookies and such).
         // We do this so that when the player requests for the ts segment, we fetch the media file again
         // and at every 5 segment interval we use the media playlist to prefetch 5 more segments
         (&Method::GET, url_path) if url_path.ends_with(".m3u8") => {
-
             let mut be_resp = req.send(PRIMARY_BACKEND)?;
             let url = Url::parse(&req_clone.get_url().to_string())?;
-            let mut path_segments = url.path_segments().ok_or_else(|| "cannot be base").unwrap();
+            let path_segments = url.path_segments().ok_or_else(|| "cannot be base").unwrap();
             let path_segment_copy = path_segments.clone();
             let file = path_segment_copy.last().unwrap().clone();
-            let query_str = "?".to_owned()  + file;
+            let query_str = "?".to_owned() + file;
 
             let len = path_segments.clone().count();
-            let mut base_url = req_clone.get_url().host_str().unwrap();
+            let base_url = req_clone.get_url().host_str().unwrap();
             let mut path_url_vec = vec![];
             path_url_vec.push(base_url);
-            for (i, path) in path_segments.enumerate().filter(|&(index, path)| index < len-1) {
+            for (_i, path) in path_segments
+                .enumerate()
+                .filter(|&(index, _path)| index < len - 1)
+            {
```

Comment:
I prefixed both `i` and `path` with `_` because the compiler is warning that these aren't being used

Reply:
I'm wondering if it's worth putting a small code comment here explaining what this block is doing